storage/replica.go

```diff
@@ -594,10 +592,10 @@ func (r *Replica) addAdminCmd(ctx context.Context, args proto.Request) (proto.Re
 
 	switch tArgs := args.(type) {
 	case *proto.AdminSplitRequest:
-		resp, err := r.AdminSplit(*tArgs)
+		resp, err := r.AdminSplit(*tArgs, r.Desc())
```

Comment:
Shouldn't this get passed in as part of the AdminSplitRequest? Otherwise, you still have a race condition. You want to be sure the split proceeds only if the range descriptor is unchanged from its value when the split was decided. Same for merge of course.